schema/Readme.md

```diff
@@ -633,6 +640,30 @@ model Block {
 }
 ```
 
+Underneath:
+
+```sql
+create table Document (
+  projectID text,
+  revision integer,
+);
+
+create table Block (
+  id integer primary key,
+  document_projectID text not null,
+  document_revision  integer not null,
+  foreign key (document_projectID, document_revision) references Document (projectID, revision)
+);
+```
+
+**Open Question** There is still magic here. How can we make `document_projectID` and `document_revision` less opinionated?
+
+###### Rules for Referencing Primary Composite Keys
+
+1. You **must** specify the relationship to the foreign fields using the `@relation` attribute. `@relation(references: [projectID, revision])`
```

Comment:
I don't know whether this in scope for this PR but in this case `@relation` must contain information about the underlying columns holding the foreign key. E.g. `@relation(from: [document_projectID, document_revision], references: [projectID, revision])`. Your suggested column names could be the default that does not require typing those names out. But it will be required with existing databases. Please talk to @do4gr who knows more about this.

Reply:
Thanks for the info, will talk to @do4gr